File: c#/src/v1/models/UserData.cs
using System;
using System.Text.Json.Serialization;

namespace Comms.Models;

public class UserData{
    [JsonPropertyName("username")]
    public string Username { get; set; }
    [JsonPropertyName("password")]
    public string ApiKey { get; set; }

    public UserData(string userName, string apiKey)
    {
        Username = userName ?? throw new ArgumentNullException(nameof(userName));
        ApiKey = apiKey ?? throw new ArgumentNullException(nameof(apiKey));
    }
}

File: js/src/v1/models/UserData.ts
export class UserData {
    public username: string;
    public apikey: string;

    constructor(userName: string, apiKey: string) {
        this.username = userName;
        this.apikey = apiKey;
    }

    public toArray(): object {
        return {
            username: this.username,
            password: this.apikey,
        };
    }
}